services/venice-server/src/main/java/com/linkedin/venice/listener/ServerStoreAclHandler.java

```diff
@@ -58,11 +58,15 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
   }
 
   /**
-   * In Venice Server, the resource name is actually a Kafka topic name.
+   * In Venice Server, the resource name is actually a Kafka topic name for STORAGE/COMPUTE but store name for DICTIONARY.
    */
   @Override
   protected String extractStoreName(String resourceName) {
-    return Version.parseStoreFromKafkaTopicName(resourceName);
+    if (Version.isVersionTopic(resourceName)) {
+      return Version.parseStoreFromKafkaTopicName(resourceName);
+    } else {
+      return resourceName;
+    }
```

Comment:
This kind of extraction is problematic since store names could also have `_vN` suffix. Can we have a check by also using the `QueryAction`? That way we would be stricter in what kind of values we allow.

I don't remember exactly where we saw this in the past, but I remember fixing this bug in another place. I'll see if I can find that place

Reply:
It was this one: https://github.com/linkedin/venice/commit/8c8b6563688ee1583a84aa25515493130a0b3525

Reply:
This is a valid concern, so maybe we can pass the `QueryAction` to this function: `extractStoreName`, and for `DICTIONARY` request, it will use the passed `resourceName` as `storeName` directly.

Reply:
Thanks for pointing this out. Will add a test for this as well.